### ISSUE TITLE
False positive in dry mode with FixNamespaceComments and NamespaceMacros

### ISSUE THREAD

**Issue by Neverlord** [llvm/llvm-project#63795]
When using the `FixNamespaceComments` option with custom `NamespaceMacros`, `clang-format -n` reports an error for code formatted by Clang-Format. Here's a full setup to reproduce:

```console
$ cat .clang-format 
---
BasedOnStyle: LLVM
FixNamespaceComments: true
Language: Cpp
NamespaceMacros: ["SUITE"]
Standard: Cpp11
UseTab: Never
...
```

```console
$ cat test.cpp 
SUITE("this.is.my.suite") {

TEST("this.is.my.test") {
  ASSERT(1 == 1);
  ASSERT(2 == 2);
}

} // SUITE("this.is.my.suite")
```

```console
$ clang-format -i test.cpp 
$ clang-format -n test.cpp 
test.cpp:8:3: warning: code should be clang-formatted [-Wclang-format-violations]
} // SUITE("this.is.my.suite")
  ^
```

/edit: tested on Clang-Format 15 and Clang-Format 16.

**Comment by owenca** [comment]
https://reviews.llvm.org/D157568
